pkg/endpoint/bpf.go

```diff
@@ -1001,9 +994,6 @@ func (e *Endpoint) syncPolicyMap() error {
 			if err != nil {
 				e.getLogger().WithError(err).Errorf("Failed to add PolicyMap key %s %d", policyKeyToPolicyMapKey.String(), entry.ProxyPort)
 				errors = append(errors, err)
-			} else {
```

Comment:
Why is this being removed? How will be track the realized state of the PolicyMap for the endpoint ?

Reply:
After the policy update is complete, the current policy is exactly the one described by the `desiredPolicy`. Instead of modifying the `realizedPolicy` contents, we simply set the `realizedPolicy` to point to `desiredPolicy`.

Reply:
This also handles the otherwise needed revert if any errors happen here.